app/javascript/components/ItemCard.jsx

```diff
@@ -37,7 +37,11 @@ export default (props) => {
 			switch (response.status) {
 				case 201: {
 					const responseBody = await response.json();
-					setQuantity(responseBody.order_items[0].doughnut.quantity)
+					setQuantity(responseBody.order_items[0].doughnut.quantity);
+					props.setOrders((prevState) => {
+						return [...prevState, responseBody];
+					});
+					props.refreshCart();
```

Comment:
Why do we need to refresh the cart on buy now?

Reply:
I did it to account for the case where you have some quantity of a doughnut in the cart, but using the "buy now" feature would cause there to not be enough inventory to checkout your cart.

Example:
* There are 15 DDoughnuts in stock
* You add 10 of them to your shopping cart
* You also buy 10 using "Buy Now"
* Now there are only 5 DDoughnuts left. Refreshing the cart displays the warning message to the user.

I'm sure there's definitely a more elegant / efficient solution to this but this was the solution that required the fewest lines of code.

Reply:
Ah gotcha that makes sense